python_scripts/trees.py

```diff
@@ -340,8 +340,11 @@ def plot_decision_function(X, y, clf, ax=None):
 #
 # Therefore, the entropy will be maximum when the proportion of samples from
 # each class is equal (i.e. $p(X_k)$ is 50%) and minimum when only samples for
-# a single class is present (i.e., $p(X_k)$ is 100%, definitely class `X`,
-# or 0%, definitely the other class).
+# a single class is present (i.e., $p(X_k)$ is 100%, only class `X`,
```

Comment:
I am not sure  to what refer the notation $p(X_k)$.
Is X the data ? I guess 'k' is the class.
Maybe we could use p_k to represent the frequency of the class k in the partition.

Reply:
Whoops, I was trying to use the same notation as the graph: https://upload.wikimedia.org/wikipedia/commons/2/22/Binary_entropy_plot.svg
I have amended it to be the same notation as the graph

Reply:
We should be careful having the same notation with the entropy formula above

Reply:
Happy to make it the same as the formula above. The notation of the graph won't be the same though

Reply:
Wait i did use the same notation as above? 
```
The entropy is defined as: $H(X) = - \sum_{k=1}^{K} p(X_k) \log p(X_k)$
```

Reply:
Did you change the notation?

Reply:
> Did you change the notation?

Yes ^^
I replace p(X_k) by p_k for the probability of class k.

Reply:
Sure, I've amended but now it doesn't match notation in the graph